File: java/json-transform/src/main/java/co/nlighten/jsontransform/functions/TransformerFunctionObject.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.functions.common.FunctionContext;
import co.nlighten.jsontransform.functions.common.TransformerFunction;

/*
 * For tests
 * @see TransformerFunctionObjectTest
 */
public class TransformerFunctionObject extends TransformerFunction {
    public TransformerFunctionObject() {
        super();
    }
    @Override
    public Object apply(FunctionContext context) {
        var streamer = context.getJsonElementStreamer(null);
        var adapter = context.getAdapter();
        var result = adapter.createObject();
        if (streamer != null) {
            streamer.stream().forEach(entry -> {
                if (adapter.isJsonArray(entry)) {
                    var size = adapter.size(entry);
                    if (size > 1) {
                        var key = adapter.get(entry, 0);
                        if (!adapter.isNull(key)) {
                            adapter.add(result, context.getAsString(key), adapter.get(entry, 1));
                        }
                    }
                }
            });
        }
        return result;
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionObject.ts
import TransformerFunction from "./common/TransformerFunction";
import FunctionContext from "./common/FunctionContext";
import { getAsString, isNullOrUndefined } from "../JsonHelpers";

class TransformerFunctionObject extends TransformerFunction {
  constructor() {
    super({});
  }

  override async apply(context: FunctionContext): Promise<any> {
    const streamer = await context.getJsonElementStreamer(null);
    const result: Record<string, any> = {};
    if (streamer != null) {
      await streamer.stream().forEach(entry => {
        if (Array.isArray(entry)) {
          const size = entry.length;
          if (size > 1) {
            const key = entry[0];
            if (!isNullOrUndefined(key)) {
              result[getAsString(key)] = entry[1];
            }
          }
        }
      });
    }
    return result;
  }
}

export default TransformerFunctionObject;
